packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -164,38 +174,77 @@ export default function lwc(pluginOptions: RollupLwcOptions = {}): Plugin {
+                    // remove query params
+                    const { filename: importeeNormalizedFilename } =
+                        parseDescriptorFromFilePath(importeeAbsPath);
 
-                const { filename } = parseQueryParamsForScopedOption(absPath);
-                absPath = filename; // remove query param
+                    if (
+                        isImplicitHTMLImport(importeeNormalizedFilename, importerFilename) &&
+                        !fs.existsSync(importeeNormalizedFilename)
+                    ) {
+                        return IMPLICIT_DEFAULT_HTML_PATH;
+                    }
 
-                if (isImplicitHTMLImport(absPath, importer) && !fs.existsSync(absPath)) {
-                    return IMPLICIT_DEFAULT_HTML_PATH;
-                }
+                    if (
+                        isImplicitCssImport(importeeNormalizedFilename, importerFilename) &&
+                        !fs.existsSync(importeeNormalizedFilename)
+                    ) {
+                        return IMPLICIT_DEFAULT_CSS_PATH;
+                    }
 
-                if (isImplicitCssImport(absPath, importer) && !fs.existsSync(absPath)) {
-                    return IMPLICIT_DEFAULT_CSS_PATH;
-                }
+                    if ((importeeExt === '.html' || importeeExt === '.css') && importerSpecifier) {
+                        // html and css files should inherit the namespace and name of the module
+                        // importing them
+                        return appendAliasSpecifierQueryParam(importeeAbsPath, importerSpecifier);
+                    }
 
-                return pluginUtils.addExtension(normalizedPath, ext);
-            } else if (importer) {
-                // Could be an import like `import component from 'x/component'`
-                try {
-                    return resolveModule(importee, importer, {
-                        modules,
-                        rootDir,
-                    }).entry;
-                } catch (err: any) {
-                    if (err && err.code !== 'NO_LWC_MODULE_FOUND') {
-                        throw err;
+                    return importeeAbsPath;
+                } else {
+                    // Could be an import like `import component from 'x/component'`
+                    try {
+                        const { entry, specifier, type } = resolveModule(importee, importer, {
+                            modules,
+                            rootDir,
+                        });
+
+                        if (type === RegistryType.alias) {
+                            // specifier must be in in namespace/name format
+                            const hasValidSpecifier = specifier.split('/').length > 1;
+                            if (hasValidSpecifier) {
+                                return appendAliasSpecifierQueryParam(entry, specifier);
+                            }
```

Comment:
It seems odd there to just ignore the specifier if it's not in the right format. Can we throw an error? And test for it? Or it is impossible to hit the `else` code path?

Reply:
I tried to only append a specifier when it was an alias, otherwise use the folder structure for the namespace / name.  

I have an example [here](https://github.com/salesforce/lwc/blob/a4699e106004f1b20b45eec27307a748d9c4a08c/packages/%40lwc/rollup-plugin/src/__tests__/resolver/resolver.spec.ts#L187-L202).

The reason for appending the alias specifier is if you had for example an alias like `x/foo` that resolved to folder path `z/bar`.  I would expect the namespace / name combo for the component to be "x / foo" and not "z / bar".

In terms of the valid specifiers, sometimes the alias evaluates to something valid but not in the namespace / name path like `lwc`.

In these cases, I ignore them and use the folder structure for the namespace / name but let me know if you think we should be using it for the namespace / name instead.

I'll include a test for this scenario.

Reply:
@nolanlawson I updated the test case in the latest [commit](https://github.com/salesforce/lwc/pull/3338/commits/47973e00fab660af8a3cbd67a5b74c3dbd234b28#diff-7a0cd956df353722bbc29288a0960fd680b602f56b483cf85c85e2524103b972R203-R208) to cover a case when the alias isn't in the namespace/name format.